### DISCUSSION TITLE
How to customize selections points of objects

### DISCUSSION THREAD

**Discussion by Clemweb** [fabricjs/fabric.js#10192] [Q&A]
Hello,

Somebody can tell me how to customize corners like this screen (specialy the rounded rectangle)

![corners](https://github.com/user-attachments/assets/56434e34-3449-4bf1-85d5-bbbc96b08ae8)

I was able to do this with v5 but with v6 something has change with _renderControls.

Thank fo help

**Comment by Clemweb** [comment]
@asturur ok I write this : 

`Control.prototype.render = function(ctx, left, top, styleOverride, fabricObject)
        {
            const size = 10; // Taille du handler
            const radius = 5; // Rayon des coins du rectangle
            ctx.save();
            ctx.translate(left, top);
            ctx.fillStyle = styleOverride.cornerColor || fabricObject.cornerColor;

            ctx.beginPath();
            ctx.moveTo(-size + radius, -size);
            ctx.lineTo(size - radius, -size);
            ctx.quadraticCurveTo(size, -size, size, -size + radius);
            ctx.lineTo(size, size - radius);
            ctx.quadraticCurveTo(size, size, size - radius, size);
            ctx.lineTo(-size + radius, size);
            ctx.quadraticCurveTo(-size, size, -size, size - radius);
            ctx.lineTo(-size, -size + radius);
            ctx.quadraticCurveTo(-size, -size, -size + radius, -size);
            ctx.closePath();

            ctx.fill();

            // Dessiner la bordure
            ctx.lineWidth = 1; // Épaisseur de la bordure
            ctx.strokeStyle = '#000000'; // Couleur de la bordure
            ctx.stroke();

            ctx.restore();
        };`

It's working, but how to know if it draw the top left corner ? or middle right ?

**Comment by asturur** [comment]
There is a duplicate of this request in another discussion. I m closing this one.
